authors/apps/notifications/tests/test_create_and_fetch.py

```diff
@@ -0,0 +1,94 @@
+        #create notification object
+        self.notify_title = "Justo followed you"
+        self.notifications = Notification(title=self.notify_title)
+        self.token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.\
+            eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.\
+            SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"
+
+class TestNotificationModel(TestSetup):
+    """Tests for the Notification Model"""
+    def test_object_creation(self):
+        """Test api can create a notification object"""
+        initial_count = Notification.objects.count()
+        self.notifications.save()
+        final_count = Notification.objects.count()
+        self.assertEqual(final_count - initial_count, 1)
+
+class TestNotificationView(TestSetup):
+    """Tests for the Notification views"""
+    def test_fetch_notifications(self):
+        """Test user can get all notifications"""
+        #ensure user is logged in first
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+        response = self.client.get(
+            "/api/notifications/", format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+    
+    def test_fetch_one_notification(self):
+        """Test user can get single notification"""
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+        response = self.client.get(
+            "/api/notifications/1", format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+    
+    def test_fetch_unauthorized_notification(self):
+        """Test user cannot get notification unauthorized"""
+        response = self.client.get(
+            "/api/notifications/", format="json")
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_fetch_article_notifications(self):
+        """Test user can get all article notifications"""
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+        response = self.client.get(
+            "/api/notifications/articles", format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_fetch_comment_notifications(self):
+        """Test user can get all comment notifications"""
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+        response = self.client.get(
+            "/api/notifications/comments", format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+    
+    def test_fetch_profile_notifications(self):
+        """Test user can get all profile notifications"""
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+        response = self.client.get(
+            "/api/notifications/profile", format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+    
+    def test_fetch_nonexist_notification(self):
+        """Test user can get nonexisting notification"""
```

Comment:
Should a user get notifications which are not existing

Reply:
Is a user in a position to distinguish read from unread notifications?

Reply:
> Should a user get notifications which are not existing

I looked at it from the backend perspective for this case, there is an endpoint to get specific notifications, for instance, `api/notifications/1` and I saw it made sense to test for 404 in case someone tries to access a notification endpoint that doesn't exist.

Reply:
> Is a user in a position to distinguish read from unread notifications?

Yes, Leo is working on tests for update action from read to unread.